hoomd/md/IntegratorTwoStep.cc

```diff
@@ -75,21 +75,25 @@ void IntegratorTwoStep::update(uint64_t timestep)
 #ifdef ENABLE_MPI
     if (m_sysdef->isDomainDecomposed())
         {
+        // Update the rigid body consituent particles before communicating so that any such
+        // particles that move from one domain to another are migrated.
+        updateRigidBodies(timestep + 1);
+
         // perform all necessary communication steps. This ensures
         // a) that particles have migrated to the correct domains
         // b) that forces are calculated correctly, if ghost atom positions are updated every time
         // step
-
-        // also updates rigid bodies after ghost updating
         m_comm->communicate(timestep + 1);
         }
     else
 #endif
-        {
+
+        // Update rigid body constituent particles after communicating (or once in serial) to ensure
+        // that all ghost constituent particle positions are set in accordance with any just
+        // communicated ghost and/or migrated rigid body centers.
         updateRigidBodies(timestep + 1);
```

Comment:
Shouldn't be there be a set of open and closed brackets here after the `else`? This is in one other place in this PR too. I'm surprised a linter didn't find this.

Reply:
Yes, will fix.